### PR TITLE
feat(console): Console for runtime wifi configuration

### PR DESCRIPTION
**Pull request by espressif-abhikroy**
Added a component that offers a console with a command that enables runtime wifi configuration for any example project.

### CODE REVIEW

File: components/console_cmd_wifi/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    // To register only wifi command skip calling console_cmd_all_register()
+    ESP_ERROR_CHECK(console_cmd_wifi_register());
+
+    ESP_ERROR_CHECK(console_cmd_start());    // Start console
+    ```
+
+### Adding a plugin command or component:
+To add a plugin command or any component from IDF component manager into your project, simply include an entry within the `idf_component.yml` file.
+
+For more details refer [IDF Component Manager](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/tools/idf-component-manager.html)
```
Comment (david-cermak) [inline components/console_cmd_wifi/README.md:39]: Isn't this a bit off topic for the description of this component? The `README` should mention the basics on this (==wifi command) component, but this paragraph describes how a(nother) component could be added. I would just delete this part.

File: components/console_cmd_wifi/console_wifi.c [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    int start_index;
+    char *help;
+} wifi_op_t;
+
+static esp_err_t wifi_help_op(wifi_op_t *self, int argc, char *argv[]);
+static esp_err_t wifi_show_op(wifi_op_t *self, int argc, char *argv[]);
+static esp_err_t wifi_sta_join_op(wifi_op_t *self, int argc, char *argv[]);
+static esp_err_t wifi_sta_leave_op(wifi_op_t *self, int argc, char *argv[]);
+//static esp_err_t wifi_sta_stop_op(wifi_op_t *self, int argc, char *argv[]);
+//static esp_err_t wifi_ap_op(wifi_op_t *self, int argc, char *argv[]);
```
Comment (david-cermak) [inline components/console_cmd_wifi/console_wifi.c:47]: Remove

File: components/console_cmd_wifi/console_wifi.c [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+
+    memset(ap_info, 0, sizeof(ap_info));
+
+    wifi_init();
+
+    esp_wifi_scan_start(NULL, true);
+    ESP_ERROR_CHECK(esp_wifi_scan_get_ap_records(&number, ap_info));
+    ESP_ERROR_CHECK(esp_wifi_scan_get_ap_num(&ap_count));
+
+    printf("\n");
```
Comment (david-cermak) [inline components/console_cmd_wifi/console_wifi.c:143]:
```suggestion
    ESP_LOGI(TAG, "\n");
```

File: components/console_cmd_wifi/console_wifi.c [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+    }
+
+    if (!strcmp("sta", argv[self->start_index + 1])) {
+        {
+            wifi_config_t wifi_config = { 0 };
+
+            wifi_init();
+            ESP_ERROR_CHECK(esp_wifi_get_config(WIFI_IF_STA, &wifi_config));
+
+            printf("\n");
```
Comment (david-cermak) [inline components/console_cmd_wifi/console_wifi.c:166]: same here

File: components/console_cmd_wifi/console_wifi.c [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+    return ESP_OK;
+}
+
+#if 0
+static esp_err_t wifi_ap_op(wifi_op_t *self, int argc, char *argv[]);
+{
+
+    return ESP_OK;
+}
+#endif
```
Comment (david-cermak) [inline components/console_cmd_wifi/console_wifi.c:234]: Should this be removed, too?

File: components/console_cmd_wifi/console_wifi.c (hunk 3 above)
Comment (espressif-abhikroy) [reply components/console_cmd_wifi/console_wifi.c:143]: The new line was added to create a gap between the network list and the wifi initialization message the appears the first time `wifi show network` command is run. But now it doesn't seem necessary. Removing the print.

File: .github/workflows/publish-docs-component.yml [hunk 6]
```diff
@@ -96,5 +96,6 @@ jobs:
             components/console_simple_init;
             components/console_cmd_ping;
             components/console_cmd_ifconfig;
+            components/console_cmd_wifi;
```
Comment (david-cermak) [inline .github/workflows/publish-docs-component.yml:99]: This will publish the new component.

Either remove this line or add a bump commit

**Review by david-cermak — APPROVED** [review]

**Pull request merged**